src/solvers/dgsem_tree/indicators_1d.jl

```diff
@@ -24,84 +24,69 @@ function create_cache(typ::Type{IndicatorHennemannGassner}, mesh, equations::Abs
 end
 
 
-function (indicator_hg::IndicatorHennemannGassner)(u, mesh::Union{TreeMesh{1}, StructuredMesh{1}},
-                                                   equations, dg::DGSEM, cache;
-                                                   kwargs...)
+# Use this function barrier and unpack inside to avoid passing closures to Polyester.jl
+# with @batch (@threaded).
+# Otherwise, @threaded does not work here with Julia ARM on macOS.
+# See https://github.com/JuliaSIMD/Polyester.jl/issues/88.
+@inline function calc_element_alpha_hg!(indicator_hg, u::AbstractArray{<:Any,3},
+                                        mesh, equations, dg, cache,
+                                        threshold, parameter_s, element)
```

Comment:
```suggestion
@inline function calc_element_alpha_hg!(indicator_hg, threshold, parameter_s, u,
                                        element, mesh::AbstractMesh{1}, equations, dg, cache)
```
I think we decided that it's easier to read when dispatching on the dimension of the `mesh` instead of `u`. The other suggestions are
- put `element` directly before `mesh`, following our `flux_differencing_kernel!` etc.
- put `threshold, parameter_s` close to the `indicator_hg` since it belongs logically to it

If you agree, please update the other files accordingly.